### DISCUSSION TITLE
Bring back option to reuse existing MikroORM instance

### DISCUSSION THREAD

**Discussion by askmrsinh** [nextauthjs/next-auth#8921] [Ideas]
### Goals

1. Avoid multiple ORM instance for same DB/schema
2. Avoid possible duplicate connection to DB, exhausting connection pool
3. Increase performance by reusing [inbuilt result caching](https://mikro-orm.io/docs/caching) on existing instance
4. This is consistent with other authjs db adapters that require an instance (e.g. [fauna](https://authjs.dev/reference/adapter/fauna), [sequelize](https://authjs.dev/reference/adapter/sequelize), [mongodb](https://authjs.dev/reference/adapter/mongodb) to name a few)

### Background

PR https://github.com/nextauthjs/adapters/pull/335 mentions that using existing MikroORM instance made things 'flaky' but provides no details on how so. Changes were later merged as https://github.com/nextauthjs/next-auth/pull/4008 which removed the ability to pass existing instance and instead, provide config.
I argue that using multiple MikroORM instances within the same app/request is even more problematic. In most use-cases the DB is used for more than just auth and the developer will have to create another instance of MikroORM to do so. Since MikroORM utilizes [IdentityMap](https://mikro-orm.io/docs/identity-map), and as per my understanding, those are unique to each MikroORM instance (i.e. there's no global Identity Map), this leads to bugs, degraded performance, and larger memory footprint.

cc: @boredland

### Proposal

I am interested in contributing.

**Comment by boredland** [comment]
The primary reason to remove it, was to simplify it and that I wasn't even sure there are performance benefits, at least when using `RequestContext` on all your API routes.
